sources/academy/tutorials/node_js/multiple-runs-scrape.md

```diff
@@ -0,0 +1,228 @@
+<TabItem value="main.ts" label="main.ts">
+
+```typescript
+import { Actor, ApifyClient, log } from 'apify';
+
+interface Input {
+    parallelRunsCount: number;
+    targetActorId: string;
+    targetActorInput: Record<string, any>;
+    targetActorRunOptions: Record<string, any>;
+}
+
+await Actor.init();
+
+const {
+    parallelRunsCount= 1,
+    targetActorId,
+    targetActorInput = {},
+    targetActorRunOptions = {},
+} = await Actor.getInput<Input>() ?? {} as Input;
+const apifyClient = new ApifyClient();
+
+if (!targetActorId) throw new Error('Missing the "targetActorId" input!');
+```
+
+</TabItem>
+</Tabs>
+
+### Reusing dataset and request queue
+
+The Master Actor will reuse its default dataset and request queue. The dataset stores the results of the scraping process, and the request queue is used as shared storage for processing requests.
+
+```typescript
+import { Actor } from 'apify';
+
+const requestQueue = await Actor.openRequestQueue();
+const dataset = await Actor.openDataset();
+```
+
+
+### State
+
+The Master Actor will maintain the state of the scraping runs to track progress and manage continuity. It will record the state of Actor runs, initializing this tracking with the first run.
+This persistent state ensures that, in migration or restart (resurrection) cases, the Actor can resume the same runs without losing progress.
+
+```typescript
+import { Actor, ApifyClient, log } from 'apify';
+
+const apifyClient = new ApifyClient();
+const state = await Actor.useState<State>('actor-state', { parallelRunIds: [], isInitialized: false });
+
+if (state.isInitialized) {
+    for (const runId of state.parallelRunIds) {
+        const runClient = apifyClient.run(runId);
+        const run = await runClient.get();
+
+        // This should happen if the run was deleted or the state was incorectly saved.
+        if (!run) throw new Error(`The run ${runId} from state does not exists.`);
+
+        if (run.status === 'RUNNING') {
+            log.info('Parallel run is already running.', { runId });
+        } else {
+            log.info(`Parallel run was in state ${run.status}, resurrecting.`, { runId });
+            await runClient.resurrect(targetActorRunOptions);
+        }
+    }
+} else {
+    for (let i = 0; i < parallelRunsCount; i++) {
+        const run = await Actor.start(targetActorId, {
+            ...targetActorInput,
+            datasetId: dataset.id,
+            requestQueueId: requestQueue.id,
+        }, targetActorRunOptions);
+        log.info(`Started parallel run with ID: ${run.id}`, { runId: run.id });
+        state.parallelRunIds.push(run.id);
+    }
+    state.isInitialized = true;
+}
+```
+
+Once Actor is initialized, it launches parallel scraper runs and waits for them to complete using `Promise.all()`.
+Additionally, by registering for abort events, the Actor can terminate all parallel runs if the Coordinator Actor is stopped.
+
+<Tabs groupId="main">
+<TabItem value="main.ts" label="main.ts">
```

Comment:
nitpick, could we make sure that codetabs are in the same order throughout the whole doc? It's both for consistency sake and due to the fact that docusaurus changes tab in whole documents if it has the same value